instance/tests/integration/integration_instance.py

```diff
@@ -287,7 +288,11 @@ def test_spawn_appserver(self):
         # We don't want to simulate e-mail verification of the user who submitted the application,
         # because that would start provisioning. Instead, we provision ourselves here.
 
-        spawn_appserver(instance.ref.pk, mark_active_on_success=True, num_attempts=2)
+        appserver = spawn_appserver(instance.ref.pk, mark_active_on_success=True, num_attempts=2)
+
+        # Test accepting beta test application
+        on_appserver_spawned(None, instance=instance, appserver=appserver)
+        self.assertEqual(instance.betatestapplication_set.first().status, BetaTestApplication.ACCEPTED)
```

Comment:
Instead of modifying the existing test, create a new one that tests this explicit functionality.